src/core/a-entity.js

```diff
@@ -435,14 +435,28 @@ var proto = Object.create(ANode.prototype, {
       if (isDefault || isMixedIn) { return; }
 
       component = this.components[name];
+
+      if (!component) { return; }
+
+      // Wait for component to initialize.
+      if (!component.initialized) {
```

Comment:
Initialization is already synchronous. This is only in the case when you do appendChild/removeChild because that's inherently asynchronous.